### ISSUE TITLE
Assemble module fails with remote_src

### ISSUE THREAD

**Issue by EspadaV8** [ansible/ansible#5089]
I've been trying to get the assemble module working, but keep coming across issues. I'll send a PR that "Works For Me", but I really don't think that it's the correct solution.

**Comment by mpdehaan** [comment]
Curious if you could share the use cases that lead to failure in this case given Brian's comment on 5057 that it works for him?

**Comment by EspadaV8** [comment]
I'm quiet possibly going about things the wrong way, but I have a folder with a number of public SSH keys in for a number of users. Each user has an account on the servers and each server has a number of vhosts. The public key for the user is added to their account and for the vhosts all users public keys are added (so they can log into a personal account if they need to sudo or, normally, just log in to the vhost user account).

At the moment I have to manually combine all the SSH keys into a 'combined' file and use that in authorised_key (using lookup('file', ...)). I've been trying to get assemble to do this for me and so I have the very simple task like this

```
- name: Assemble SSH public key file
  assemble:
    remote_src: False
    src: "/Users/andrew/Workspace/ansible-demo/private/ansible/roles/common/files/ssh/keys"
    dest: '/tmp/ssh_combined'
```

Without the PR #5091 I get a number of errors about various different things (basically what he commits mention). After the PR is applied I correctly get the /tmp/ssh_combined file created on the remote host (I'm now looking at how to use lookup on a remote file or a way to save the file to a local folder).

If I am doing something completely wrong, just let me know, but I still believe that there must be something wrong with the assemble for me to get the errors I was seeing.

**Comment by jctanner** [comment]
possibly fixed by #5091. Need to validate and close this ticket.

**Comment by EspadaV8** [comment]
@jctanner Yeah, this should be fixed by #5091. I opened this in case someone else had a better fix than my PR and could reference the original issue.
